### PR TITLE
feat: data intake module

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Testing

Before you submit a pull request, please make sure you have to following:

- [x] Your code contains tests that cover all new code and changes
- [x] All new and existing tests passed
- [x] I have read the [Contributing Guidelines](https://github.com/aws-solutions-library-samples/guidance-for-overhead-imagery-inference-on-aws/blob/main/CONTRIBUTING.md) and agree to follow the [Code of Conduct](
https://github.com/aws-solutions-library-samples/guidance-for-overhead-imagery-inference-on-aws/blob/main/CODE_OF_CONDUCT.md))

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by edparris — COMMENTED** [review]
Comments in di-container.ts and di-dataplane.ts look like cut and paste errors from tile server.

File: bin/deploy-data-intake.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  // Deploy the Virtual Private Cloud (VPC) resources for OversightML
+  const dataplaneStack = new DIDataplaneStack(
+    app,
+    `${targetAccount.name}-DIDataplane`,
+    {
+      env: targetEnv,
+      account: targetAccount,
+      osmlVpc: vpcStack.resources,
+      dockerImageCode: containerStack.resources.dockerImageCode,
+      description: "Guidance for Overhead Imagery Inference on AWS (SO9240)"
```
Comment (RanbirAulakh) [inline bin/deploy-data-intake.ts:53]: Update the description to: `Data Intake, Guidance for Overhead Imagery Inference on AWS (SO9240)` so its consistent with Tile Server. We would need to update Model Runner description as well.

File: bin/deploy-data-intake.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  buildFromSource: boolean = true,
+) {
+  const containerStack = new DIContainerStack(
+    app,
+    `${targetAccount.name}-DIContainer`,
+    {
+      env: targetEnv,
+      account: targetAccount,
+      osmlVpc: vpcStack.resources,
+      buildFromSource: buildFromSource
```
Comment (RanbirAulakh) [inline bin/deploy-data-intake.ts:38]: Missing description here.

File: bin/deploy-data-intake.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+import { App, Environment } from "aws-cdk-lib";
+import { OSMLAccount } from "osml-cdk-constructs";
+
+import { DIContainerStack } from "../lib/osml-stacks/data_intake/di-container";
+import { DIDataplaneStack } from "../lib/osml-stacks/data_intake/di-dataplane";
+import { OSMLVpcStack } from "../lib/osml-stacks/osml-vpc";
+
+/**
+ * Deploys the image handler service.
```
Comment (RanbirAulakh) [inline bin/deploy-data-intake.ts:13]: Would this be `Deploys the data intake service`?

File: bin/app.ts [hunk 4]
```diff
@@ -56,6 +57,11 @@ if (targetAccount.deployTileServer) {
   deployTileServer(app, targetEnv, targetAccount, vpcStack, buildFromSource);
 }
 
+// Deploy the image intake application within the same VPC.
```
Comment (RanbirAulakh) [inline bin/app.ts:60]: update `image intake` to `data intake`

**Comment by RanbirAulakh** [comment]
Also, one more comment which I have noticed in the Github workflow. Could you take the time to update this line to include the newly created Dataplane for intake component? 
- https://github.com/aws-solutions-library-samples/guidance-for-processing-overhead-imagery-on-aws/blob/dev/scripts/dev_destroy.sh#L43

File: bin/deploy-data-intake.ts (hunk 1 above)
Comment (drduhe) [reply bin/deploy-data-intake.ts:53]: Fxed

File: bin/deploy-data-intake.ts (hunk 3 above)
Comment (drduhe) [reply bin/deploy-data-intake.ts:13]: Fixed

File: bin/app.ts (hunk 4 above)
Comment (drduhe) [reply bin/app.ts:60]: fixed

File: bin/deploy-data-intake.ts (hunk 2 above)
Comment (drduhe) [reply bin/deploy-data-intake.ts:38]: fixed

**Pull request merged**